src/util/arith_tools.cpp

```diff
@@ -111,6 +111,29 @@ bool to_integer(const constant_exprt &expr, mp_integer &int_value)
 
 /*******************************************************************\
 
+Function: to_unsigned_integer
+
+  Inputs: a constant expression and a reference to an unsigned int
+
+ Outputs: an error flag
+
+ Purpose: convert a positive integer expression to an unsigned int
+
+\*******************************************************************/
+
+bool to_unsigned_integer(const constant_exprt &expr, unsigned &uint_value)
+{
+  mp_integer i;
+  if(to_integer(expr, i))
+    return true;
+  if(i<0)
+    return true;
+  else
+    return integer2unsigned(i);
```

Comment:
yes sorry, corrected it https://github.com/diffblue/cbmc/pull/374/commits/d8f28260a292d5bad35671803c54e4f5386fe349
For tests, this pull request does not include all the parts of the string solver so we cannot directly test it.